Saving Functionality for Words in Study Tab

PFKimmerle opened the pull request:
Fixes Sub-issue #273 

## Summary
**Frontend**
- Users can click add "+" button on word popup in Study tab
- Connect button to save-word API endpoint
- Show "✓" checkmark when word is successfully saved

**Backend**
- Create endpoint to save a word to the user's account
- Return 409 to handle error cases (duplicate words)

![1](https://github.com/user-attachments/assets/0ff9f717-ab1e-427b-83ff-87582394cac4)
![2](https://github.com/user-attachments/assets/4a5430b7-7858-4348-bd26-da3252433130)
![3](https://github.com/user-attachments/assets/5abed03f-ce75-4dea-a30f-5f49a049ac74)

PFKimmerle committed:
Saving Functionality for Words in Study Tab

- Backend added saveWord route and sends 409 if duplicate
- Frontend "+" button calls it and shows "✓" if saved/ duplicate
- StudyTab sends textId

JPEscobari commented on server/controllers/api/textController.js:
Everything is working just fine in `api/textController.js`, but I see you are using MongoDB syntax to update the `cards` array.

Are you familiar with Mongoose methods?

Mongoose is kind of a translator between our Node.js code and the MongoDB database. It adds some user-friendly features and allows us to enforce the structure of our schema when writing documents (our cards) to the database.

We have Mongoose installed in our project and imported into our database config file, so we might as well use it for consistency.

It's nothing major, but can you try replacing line 68:
```
Text.findByIdAndUpdate(textId, { $push: { cards: card._id } });
```

with:

```
const text = await Text.findById(textId);
text.cards.push(card._id);
await text.save();
```

See if it works.

It's a little more verbose, but I think more descriptive. Give it a try and I hope you like it!

JPEscobari commented on client/src/KnowNativePage/TextPage/components/WordPopup/WordPopup.scss:
Styling is on point. We just need to make a minor adjustment to the class and class selectors to conform to the BEM naming convention we've been trying to maintain across the project.

In short, BEM (Block Element Modifier) uses double underscores for elements and double hyphens for modifiers.

In this case:

The Block is: .word-pop
The Element: __add
The Modifier: --saved

So, can you please update:
```
.word-popup__add.saved {
  background: transparent;
  color: #57bcaa;
}
```
to:
```
.word-popup__add--saved {
  background: transparent;
  color: #57bcaa;
}
```

You'll also need to make an adjustment to the `wordPopup.jsx` component, off course:
```
<div className={`word-popup__add${isSaved ? ' word-popup__add--saved' : ''}`}></div>
```

JPEscobari reviewed (changes requested):
The flow looks super solid @PFKimmerle .

The frontend collects and sends all necessary data. The backend checks for duplicates, creates the card, and links it to the text.

You've also taken care of all the security details including the JWT middleware and the UI disables the save button when a word is saved or already exists.

I'm adding two comments on two separate files to see if you can add those final touches to make this commit 100% consistent with what we have accross the code base.

Check out my comments and please let me know if you have questions or suggestions as to how we should go about it.

Anyways, genius work and amazing contribution!

PFKimmerle committed:
Switch saveWord to Mongoose-style document update

- swapped MongoDB $push for Mongoose document update
- updated WordPopup class to BEM --saved

PFKimmerle commented:
Awesome, thanks for the feedback @JPEscobari
I made both updates (Mongoose-style update and BEM class rename) and pushed the changes. I also tested by adding a new word 旅程 (journey), and confirmed the DB shows card creation as expected. Let me know if there’s anything else you’d like me to adjust.

JPEscobari reviewed (approved):
Thank you so much Patricia! Another big contribution to the project!

This is ready to merge.
